libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/TripSession.kt

```diff
@@ -14,7 +14,9 @@ internal interface TripSession {
     val tripService: TripService
     val locationEngine: LocationEngine
     val locationEngineRequest: LocationEngineRequest
-    var route: DirectionsRoute?
+    var route: DirectionsRoute
```

Comment:
We shouldn't need to avoid nullability here. It's a valid state for us and Nav Native.

Reply:
https://github.com/mapbox/mapbox-navigation-android/pull/3204#issuecomment-647501993